src/utils/mtev_watchdog.c

```diff
@@ -192,6 +193,115 @@ typedef enum {
   GLIDE_WATCHDOG
 } glide_reason_t;
 
+struct manage {
+  pid_t pid;
+  struct timeval last_start;
+  uint32_t start_attempts;
+  char *file;
+  char **argv;
+  char **envp;
+  mtev_log_stream_t out;
+  mtev_log_stream_t err;
+};
+
+static struct manage *managed;
+static int nmanaged;
+static bool manage_allowed = false;
+/* This is not exposed, only called from mtev_main */
+void mtev_watchdog_allow_manage() {
+  manage_allowed = true;
+}
+void mtev_watchdog_manage(const char *file, char * const *argv, char * const *envp,
+                          mtev_log_stream_t out, mtev_log_stream_t err) {
+  int i;
+  int idx = nmanaged++;
+  mtevAssert(manage_allowed);
+  managed = realloc(managed, sizeof(*managed) * nmanaged);
+  memset(&managed[idx], 0, sizeof(*managed));
+  managed[idx].file = strdup(file);
+  for(i=0;argv[i] != NULL;i++);
+  managed[idx].argv = calloc(i+1, sizeof(char *));
+  for(i=0;argv[i] != NULL;i++) managed[idx].argv[i] = strdup(argv[i]);
+  managed[idx].argv[i] = NULL;
+  for(i=0;envp[i] != NULL;i++);
+  managed[idx].envp = calloc(i+1, sizeof(char *));
+  for(i=0;envp[i] != NULL;i++) managed[idx].envp[i] = strdup(envp[i]);
+  managed[idx].envp[i] = NULL;
```

Comment:
```suggestion
  managed[idx].envp[i+1] = NULL;
```

Reply:
that's not right. we leave the loop with [i] unset.  It should be `i` not `i+1`